Project.toml

```diff
@@ -6,30 +6,37 @@ version = "0.6.16"
 Adapt = "79e6a3ab-5dfb-504d-930d-738a2a938a0e"
 ConstructionBase = "187b0558-2788-49d3-abe0-74a17ed4e7c9"
 DataAPI = "9a962f9c-6df0-11e9-0e5d-c546b8b5ee8a"
-GPUArraysCore = "46192b85-c4d5-4398-a991-12ede77f4527"
```

Comment:
I'm a bit confused by this: how can test pass on older julia versions if we don't have these as hard dependencies?

Reply:
IIUC, `Pkg.jl` for old julia has been patched, and weakdeps would be parsed as deps there

Reply:
Still, I think it might be safer to follow the guidelines [here](https://pkgdocs.julialang.org/v1/creating-packages/#Transition-from-normal-dependency-to-extension) and have Adapt, GPUArraysCore and StaticArrays in deps as well.

Reply:
Make sense, looks like I get confused by the Requires.jl based solution.